store/file/square.go

```diff
@@ -0,0 +1,131 @@
+package file
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+type square [][]share.Share
+
+// readSquare reads Shares from the reader and returns a square. It assumes that the reader is
+// positioned at the beginning of the Shares. It knows the size of the Shares and the size of the
+// square, so reads from reader are limited to exactly the amount of data required.
+func readSquare(r io.Reader, shareSize, edsSize int) (square, error) {
+	odsLn := edsSize / 2
+
+	square := make(square, odsLn)
+	for i := range square {
+		square[i] = make([]share.Share, odsLn)
+		for j := range square[i] {
+			square[i][j] = make(share.Share, shareSize)
+		}
+	}
+
+	// TODO(@walldiss): run benchmark to find optimal size for this buffer
```

Comment:
I would park this in the issue, so that some rando interested in low-level things could contribute

Reply:
It might not be an issue to be created now, but we could make a little list of low-prior issues to be posted to GH. This one for example and the Size returning error